src/customer/strategies/bolt/bolt-customer-strategy.ts

```diff
@@ -0,0 +1,150 @@
+import { PaymentMethodActionCreator } from '../../../payment';
+import { PaymentMethodFailedError } from '../../../payment/errors';
+import { BoltCheckout, BoltScriptLoader } from '../../../payment/strategies/bolt';
+import CustomerActionCreator from '../../customer-action-creator';
+import CustomerCredentials from '../../customer-credentials';
+import { CustomerInitializeOptions, CustomerRequestOptions, ExecutePaymentMethodCheckoutOptions } from '../../customer-request-options';
+import CustomerStrategy from '../customer-strategy';
+
+export default class BoltCustomerStrategy implements CustomerStrategy {
+    private _boltClient?: BoltCheckout;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _boltScriptLoader: BoltScriptLoader,
+        private _customerActionCreator: CustomerActionCreator,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator
+    ) {}
+
+    async initialize(options: CustomerInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId, bolt } = options;
+        const { onInit } = bolt || {};
+
+        if (!methodId) {
+            throw new InvalidArgumentError('Unable to proceed because "methodId" argument is not provided.');
+        }
+
+        const state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethods());
+
+        const paymentMethod = state.paymentMethods.getPaymentMethod(methodId);
+
+        if (!paymentMethod || !paymentMethod.initializationData.publishableKey) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        const { developerConfig, publishableKey } = paymentMethod.initializationData;
+
+        this._boltClient = await this._boltScriptLoader.load(publishableKey, paymentMethod.config.testMode, developerConfig);
+
+        if (onInit && typeof onInit === 'function') {
+            const email = this._getCustomerEmail();
+            const hasBoltAccount = await this._hasBoltAccount(email);
+
+            onInit(hasBoltAccount);
+        }
+
+        return this._store.getState();
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        this._boltClient = undefined;
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    signIn(credentials: CustomerCredentials, options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        return this._store.dispatch(
+            this._customerActionCreator.signInCustomer(credentials, options)
+        );
+    }
+
+    signOut(options?: CustomerRequestOptions): Promise<InternalCheckoutSelectors> {
+        return this._store.dispatch(
+            this._customerActionCreator.signOutCustomer(options)
+        );
+    }
+
+    async executePaymentMethodCheckout(options?: ExecutePaymentMethodCheckoutOptions) {
+        const { methodId, fallback } = options || {};
+        const defaultFallback = () => {};
```

Comment:
Why not use `noop` here?

Reply:
We can use it here, thanks! 🙏

Reply:
I added `noop` as default value for `fallback` and  removed checkoutFallback variable because it is unnecessary now.